database/src/main/postgres/runs/V1.8.3__get_partitioning_checkpoint.sql

```diff
@@ -0,0 +1,105 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Function: runs.get_partitioning_checkpoints(JSONB, INT, TEXT)
+CREATE OR REPLACE FUNCTION runs.get_partitioning_checkpoints(
+    IN i_partitioning              JSONB,
+    IN i_limit                     INT DEFAULT 5,
+    IN i_checkpoint_name           TEXT DEFAULT NULL,
+    OUT status                     INTEGER,
+    OUT status_text                TEXT,
+    OUT id_checkpoint              UUID,
+    OUT checkpoint_name            TEXT,
+    OUT measure_name               TEXT,
+    OUT measure_columns            TEXT[],
+    OUT measurement_value          JSONB,
+    OUT checkpoint_start_time    TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_end_time        TIMESTAMP WITH TIME ZONE
+)
+    RETURNS SETOF record AS
+$$
+    -------------------------------------------------------------------------------
+--
+-- Function: runs.get_partitioning_checkpoints(JSONB, INT, TEXT)
+--      Returns all the checkpoint for the given partitioning and checkpoint name
+--
+-- Parameters:
+--      i_partitioning      - partitioning for requested checkpoints
+--      i_limit             - limit of the number of checkpoints to return
+--      i_checkpoint_name   - name of the checkpoint to filter by
+--
+-- Returns:
+--      status                  - Status code
+--      status_text             - Status message
+--      id_checkpoint           - ID of the checkpoint
+--      checkpoint_name         - Name of the checkpoint
+--      measure_name            - Name of the measure
+--      measure_columns         - Columns of the measure
+--      measurement_value       - Value of the measurement
+--      checkpoint_start_time   - Time of the checkpoint
+--      checkpoint_end_time     - End time of the checkpoint computation
+--
+-- Status codes:
+--      11 - OK
+--      41 - Partitioning not found
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _fk_partitioning                    BIGINT;
+BEGIN
+    _fk_partitioning = runs._get_id_partitioning(i_partitioning);
+
+    IF _fk_partitioning IS NULL THEN
+        status := 41;
+        status_text := 'Partitioning not found';
+        RETURN NEXT;
+        RETURN;
+    END IF;
+
+    RETURN QUERY
+        SELECT
+            11 AS status,
+            'Ok' AS status_text,
+            c.id_checkpoint,
+            c.checkpoint_name,
+            md.measure_name,
+            md.measured_columns,
+            m.measurement_value,
+            c.process_start_time AS checkpoint_start_time,
+            c.process_end_time AS checkpoint_end_time
+        FROM
+            runs.checkpoints c
+        JOIN
+            runs.measurements m ON c.id_checkpoint = m.fk_checkpoint
+        JOIN
+            runs.measure_definitions md ON m.fk_measure_definition = md.id_measure_definition
```

Comment:
**For future (don't fix 😉):**
I would suggest to use capitals for table aliases. It makes reading and their recognition easier.

Reply:
Yep, I fully support that idea and actually started to do that. (I even use keyword `AS` - it's my habbit)